commands/azureCommands/create-registry.ts

```diff
@@ -0,0 +1,210 @@
+
+    } catch (error) {
+        return;
+    }
+    const client = AzureCredentialsManager.getInstance().getContainerRegistryManagementClient(subscription);
+
+    let registryName: string;
+    try {
+        registryName = await acquireRegistryName(client);
+    } catch (error) {
+        return;
+    }
+
+    const sku: string = await acquireSKU();
+    location = await acquireLocation(resourceGroup, subscription);
+
+    client.registries.beginCreate(resourceGroup.name, registryName, { 'sku': { 'name': sku }, 'location': location }).then((response): void => {
+        vscode.window.showInformationMessage(response.name + ' has been created succesfully!');
+    }, (error): void => {
+        vscode.window.showErrorMessage(error.message);
+    })
+
+    //Acquiring telemetry data here
+    if (reporter) {
+        /* __GDPR__
+           "command" : {
+              "command" : { "classification": "SystemMetaData", "purpose": "FeatureInsight" }
+           }
+         */
+        reporter.sendTelemetryEvent('command', {
+            command: teleCmdId
+        });
+
+        if (registryName.toLowerCase().indexOf('azurecr.io')) {
+            /* __GDPR__
+               "command" : {
+                  "command" : { "classification": "SystemMetaData", "purpose": "FeatureInsight" }
+               }
+             */
+            reporter.sendTelemetryEvent('command', {
+                command: teleAzureId
+            });
+        }
+    }
+
+}
+async function acquireSKU(): Promise<string> {
+    let skus: string[] = ["Basic", "Standard", "Premium"];
+    let sku: string;
+    sku = await vscode.window.showQuickPick(skus, { 'canPickMany': false, 'placeHolder': 'Choose a SKU' });
+    if (sku === undefined) { throw new Error('User exit'); }
+
+    return sku;
+}
+
+async function acquireRegistryName(client: ContainerRegistryManagementClient): Promise<string> {
+    let opt: vscode.InputBoxOptions = {
+        ignoreFocusOut: false,
+        prompt: 'Registry name? '
+    };
+    let registryName: string = await vscode.window.showInputBox(opt);
+
+    let registryStatus: RegistryNameStatus = await client.registries.checkNameAvailability({ 'name': registryName });
+    while (!registryStatus.nameAvailable) {
+        opt = {
+            ignoreFocusOut: false,
+            prompt: "That registry name is unavailable. Try again: "
+        }
+        registryName = await vscode.window.showInputBox(opt);
+
+        if (registryName === undefined) { throw new Error('user Exit'); }
+        registryStatus = await client.registries.checkNameAvailability({ 'name': registryName });
+    }
+    return registryName;
+}
+
+// INPUT HELPERS
+async function acquireSubscription(): Promise<SubscriptionModels.Subscription> {
+    const subs = AzureCredentialsManager.getInstance().getFilteredSubscriptionList();
+
+    if (subs.length === 0) { vscode.window.showErrorMessage('You do not have any subscriptions. Head over to Azure Portal to make one.'); }
```

Comment:
>Head over to Azure Portal to make one. [](start = 96, length = 38)

Visit [https://portal.azure.com](https://portal.azure.com) ?

Reply:
Added a button that takes you there